packages/block-editor/src/components/story-update-progress/README.md

```diff
@@ -0,0 +1,105 @@
+StoryUpdateProgress
+===================
+
+`StoryUpdateProgress` shows a progress bar while the media files associated with a Story block are being saved and uploaded
+
+## Usage
+
+Usage example
+
+```jsx
+import { ImageBackground, Text, View } from 'react-native';
+import {
+	StoryUpdateProgress,
+} from '@wordpress/block-editor';
+
+function StoryProgress( { url, id } ) {
+	return (
+		<StoryUpdateProgress
+			mediaId={ id }
+			renderContent={ ( { isSaveFailed, retryMessage } ) => {
+				return (
+					<ImageBackground
+						resizeMethod="scale"
+						source={ { uri: url } }
+					>
+						{ isSaveFailed &&
+							<View>
+								<Text>{ retryMessage }</Text>
+							</View>
+						}
+					</ImageBackground>
+				);
+			} }
+		/>
+	);
+}
+```
+
+## Props
+
+### mediaFiles
+
+A collection of media ID that identifies the current story upload.
+
+- Type: `Array`
+- Required: Yes
+- Platform: Mobile
+
+### renderContent
+
+Content to be rendered along with the progress bar, usually the thumbnail of the media being uploaded.
+
+- Type: `React components`
+- Required: Yes
+- Platform: Mobile
+
+It passes an object containing the following properties:
+
+`{ isUploadInProgress, isUploadFailed, isSaveInProgress, isSaveFailed, retryMessage }`
+
+### onUpdateMediaProgress
+
+Callback called when the progress of the upload is updated.
+
+- Type: `Function`
+- Required: No
+- Platform: Mobile
+
+The argument of the callback is an object containing the following properties:
+
+`{ mediaId, mediaUrl, progress, state }`
+
+### onFinishMediaUploadWithSuccess
+
+Callback called when the media file has been uploaded successfully.
+
+- Type: `Function`
+- Required: No
+- Platform: Mobile
+
+The argument of the callback is an object containing the following properties:
+
+`{ mediaId, mediaServerId, mediaUrl, progress, state }`
+
+### onFinishMediaUploadWithFailure
+
+Callback called when the media file couldn't be uploaded.
+
+- Type: `Function`
+- Required: No
+- Platform: Mobile
+
+The argument of the callback is an object containing the following properties:
+
+`{ mediaId, progress, state }`
```

Comment:
@mzorz since the upload failed do we still want to pass the progress and the state? Just checking 🙏

Reply:
this part is copied from the MediaUploadProgress component (we may replace that one with this one in the long run as mentioned here https://github.com/WordPress/gutenberg/pull/26005#issuecomment-707667916)

I think we need it so it's easy to override the progress bar being updated / reset if needed (up to the caller to send 0.0 or 1.0, for example)